python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -1335,12 +1335,9 @@ def keys(self):
         if nkeys == 0:
             return cudf.core.index.as_index([], name=None)
         elif nkeys > 1:
-            return cudf.MultiIndex(
-                source_data=cudf.DataFrame(
-                    dict(zip(range(nkeys), self._key_columns))
-                ),
-                names=self.names,
-            )
+            return cudf.MultiIndex._from_data(
+                dict(zip(range(nkeys), self._key_columns))
+            ).set_names(self.names)
```

Comment:
```suggestion
            result = cudf.MultiIndex._from_data(
                dict(zip(range(nkeys), self._key_columns))
            )
            result._set_names(self.names, inplace=True)
            return result
```

Should we skip going through all sorts of validations in actual `set_names`?

Reply:
Good call, that would be good yes (but we can still use the chained syntax as above).

Reply:
Yup, either way works

Reply:
```suggestion
            return cudf.MultiIndex._from_data(
                dict(zip(range(nkeys), self._key_columns))
            )._set_names(self.names)
```